hoomd/md/IntegratorTwoStep.cc

```diff
@@ -114,8 +114,11 @@ void IntegratorTwoStep::update(uint64_t timestep)
         m_prof->push("Integrate");
 
     // perform the second step of the integration on all groups
-    for (auto& method : m_methods)
+    // reversed for integrators so that the half steps will be performed symmetrically
+    // TODO: make sure this is valid for non-alchemical integrators
+    for (auto method_ptr = m_methods.rbegin(); method_ptr != m_methods.rend(); method_ptr++)
         {
+        auto method = (*method_ptr);
```

Comment:
This is probably the most concerning thing that I did. But is required for any method utilizing a Trotter factorization to separate timescales. Their halfsteps must be like bookends with the steps of the shorter timescale occurring between. This also means that our methods with this longer timescale must be first in the m_methods vector.